### PR TITLE
Mlh0079 2575

### PR DESCRIPTION
**Pull request by Michaellh0079**

### CODE REVIEW

File: dmrpp_generator/dmrpp.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    def get_https_file(self, url, host_path='/tmp'):
+        self.get_http_file(url, host_path)
+
+    def get_http_file(self, url, host_path="/tmp"):
+        """
+        Downloads a file from the provided url.
+        :param url: Location to find the file to download http or https
+        :param host_path: Where to store the downloaded file
+        """
+        filename = url.rsplit('/', 1)[-1]
```
Comment (amarouane-ABDELHAK) [inline dmrpp_generator/dmrpp.py:22]: use os.path.basename(url) to get the name of the file

File: dmrpp_generator/dmrpp.py [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+        Downloads a file from the provided url.
+        :param s3_link: Location to find the file to download from S3
+        :param host_path: Where to store the downloaded file
+        """
+        filename = s3_link.rsplit('/', 1)[-1]
+        reg_res = re.match(rf'^.*://([^/]*)/(.*)', s3_link)
+        bucket_name = reg_res.group(1)
+        key = reg_res.group(2)
+        try:
+            self.s3_client.download_file(bucket_name, key, f'{host_path}/{filename}')
```
Comment (amarouane-ABDELHAK) [inline dmrpp_generator/dmrpp.py:41]: Can we check here if the file already exist to not download it? same thing for HTTP protocol

File: dmrpp_generator/dmrpp.py (hunk 1 above)
Comment (Michaellh0079) [reply dmrpp_generator/dmrpp.py:23]: The current implementation is more performant. See the following example:
```
if __name__ == "__main__":
    t = 'https://catalog.uah.edu/grad/colleges-departments/science/earth-system-science/earth-system-science.pdf'
    iterations = 1000000
    st = time.time()
    for _ in range(iterations):
        filename = t.rsplit('/', 1)[-1]
    et1 = time.time() - st

    st = time.time()
    for _ in range(iterations):
        os.path.basename(t)
    et2 = time.time() - st

    print(f'et1: {et1}, et2: {et2}')
    pass
```

Comment (amarouane-ABDELHAK) [reply dmrpp_generator/dmrpp.py:23]: But the basepath is cleaner and more intuitive

**Review by amarouane-ABDELHAK — APPROVED** [review]

**Pull request merged**